lms/static/scripts/frontend_apps/components/EmailPreferencesApp.tsx

```diff
@@ -1,15 +1,41 @@
-import { InfoIcon, Link } from '@hypothesis/frontend-shared';
+import { InfoIcon, Link, ToastMessages } from '@hypothesis/frontend-shared';
 import classnames from 'classnames';
 import { useCallback, useState } from 'preact/hooks';
 
+import type { SelectedDays } from '../config';
 import { useConfig } from '../config';
+import { useToastMessages } from '../hooks/use-toast-messages';
 import EmailPreferences from './EmailPreferences';
 
 export default function EmailPreferencesApp() {
   const { emailPreferences } = useConfig(['emailPreferences']);
-  const [selectedDays, setSelectedDays] = useState(emailPreferences);
+  const [selectedDays, setSelectedDays] = useState(
+    emailPreferences.selectedDays
+  );
   const [saving, setSaving] = useState(false);
   const onSave = useCallback(() => setSaving(true), []);
+  const { toastMessages, dismissToastMessage } = useToastMessages(
+    emailPreferences.result
+      ? [
+          {
+            message: emailPreferences.result,
+            type: 'success',
+          },
+        ]
+      : []
+  );
+  const updateSelectedDays = useCallback(
+    (newSelectedDays: Partial<SelectedDays>) => {
+      setSelectedDays(prev => ({ ...prev, ...newSelectedDays }));
+
+      // Dismiss toast message if visible and selection actually changed
+      // compared to initial value
```

Comment:
Where is the "selection actually changed" aspect happening? Are you assuming that the callback won't be invoked unless the selection changed?

Reply:
Ugh, this comment is wrong. That was my initial intention (and apparently I wrote it down, haha), but then I changed my mind 😅

Since downstream components only call with partial objects, knowing if the value did actually change is not that straightforward.

I guess one option would be to compare every key from provided partial and check if any of them has a different value from `emailPreferences.selectedDays` (which holds the initial value).

But in the end I decided to not compare it and always dismiss the notification, for simplicity.

What would be your preference here?

Reply:
> I guess one option would be to compare every key from provided partial and check if any of them has a different value from emailPreferences.selectedDays (which holds the initial value).

Ok, I have finally done this.

This covers two edge cases, which is that all days are selected and you click `Select all`, or no days are selected and you click `Select none`. In those two cases it won't dismiss the toast message.

Reply:
> But in the end I decided to not compare it and always dismiss the notification, for simplicity.

I would have been fine with this, but if you want to do the slightly more elaborate thing that's OK too.